applications/crossbar/src/modules/cb_system_configs.erl

```diff
@@ -324,14 +255,69 @@ update_db(Context) ->
+set_id(ConfigId, Node, JObj) ->
+    kz_doc:set_id(JObj, <<ConfigId/binary, "/", Node/binary>>).
+
+maybe_set_id(ConfigId, JObj) ->
+    case kapps_config:get_category(ConfigId) of
+        {ok, Doc} -> kz_json:merge_recursive(Doc, JObj);
+        _ -> kz_doc:set_id(JObj, ConfigId)
+    end.
+
+-spec strip_id(kz_json:object()) -> kz_json:object().
+strip_id(JObj) -> kz_json:delete_key(<<"id">>, JObj, prune).
+
+-spec validate_with_parent(cb_context:context(), ne_binary(), ne_binary(), kz_json:object()) -> cb_context:context().
+validate_with_parent(Context, ConfigId, Node, Parent) ->
+    RequestData = strip_id(cb_context:req_data(Context)),
+    FullConfig = kz_json:merge_recursive(Parent, RequestData),
+    Schema = kapps_config_util:system_schema_name(ConfigId),
+    cb_context:validate_request_data(Schema, cb_context:set_req_data(Context, FullConfig),
+                                     fun(Ctx) ->
+                                             Doc = kz_json:set_value(Node, kz_json:diff(RequestData, Parent),
+                                                                     kz_doc:set_id(maybe_new(kapps_config:get_category(ConfigId)), ConfigId)),
+                                             cb_context:set_doc(Ctx, Doc)
+                                     end
+                                    ).
+
+-spec validate_request(cb_context:context(), ne_binary(), ne_binary(), kz_json:object()) -> cb_context:context().
+validate_request(Context, Id, Schema, Parent) ->
+    RequestData = strip_id(cb_context:req_data(Context)),
+    FullConfig = kz_json:merge_recursive(strip_id(kz_json:public_fields(Parent)), RequestData),
+    cb_context:validate_request_data(Schema, cb_context:set_req_data(Context, FullConfig),
+                                     fun(Ctx) ->
+                                             Doc = maybe_set_id(Id, kz_json:diff(RequestData, Parent)),
+                                             cb_context:set_doc(Ctx, Doc)
+                                     end
+                                    ).
+
+-spec maybe_new({ok, kz_json:object()}) -> kz_json:object().
+maybe_new({ok, JObj}) -> JObj;
+maybe_new(_) -> kz_json:new().
+
+default(Config) ->
+    kz_doc:set_id(kz_json_schema:default_object(kapps_config_util:system_schema(Config)), Config).
+
+-spec get_system_config(ne_binary(), ne_binary()) -> kz_json:object().
+get_system_config(Config, Node) ->
+    Default = default(Config),
+    System = maybe_new(kapps_config:get_category(Config)),
+    SystemNode = kz_json:get_value(Node, System, kz_json:new()),
+    kz_json:merge_recursive(Default, SystemNode).
+
+-spec make_schema(ne_binary()) -> kz_json:object().
+make_schema(Id) ->
+    Flat = [
+            {[<<"patternProperties">>, <<".+">>, <<"$ref">>], kapps_config_util:system_schema_name(Id)}
+           ,{[<<"patternProperties">>, <<".+">>, <<"type">>], <<"object">>}
+           ,{<<"type">>, <<"object">>}
+           ],
```

Comment:
can we chsnge this to
```
    Flat = [
            {<<"$schema">>,<<"http://json-schema.org/draft-04/schema#">>}
           ,{<<"id">>, <<"system_config">>}
           ,{<<"type">>, <<"object">>}
           ,{<<"default">>, kz_json:new()}
           ,{[<<"properties">>, <<"default">>, <<"$ref">>], kapps_config_util:system_schema_name(Id)}
           ,{[<<"properties">>, <<"default">>, <<"default">>], kz_json:new()}
           ,{[<<"properties">>, <<"default">>, <<"type">>], <<"object">>}
           ,{[<<"patternProperties">>, <<".+">>, <<"$ref">>], kapps_config_util:system_schema_name(Id)}
           ,{[<<"patternProperties">>, <<".+">>, <<"type">>], <<"object">>}
           ],
```
it will ensure the `default` object is present and with defaults.